### PR TITLE
[CORL-3120]: Add initialStatus to comment and use for pre-mod labels in moderate cards

### PR DESCRIPTION
**Pull request by kabeaty**
## What does this PR do?

These changes add an `initialStatus` field to `Comment`. This helps have a clearer history from the initial creation of the comment into what its status history is. A `status` field is added to comment revisions, so we also have status per revision now. Then we use the first revision's status to populate `initialStatus`.

## These changes will impact:

- [ ] commenters
- [x] moderators
- [x] admins
- [ ] developers

## What changes to the GraphQL/Database Schema does this PR introduce?

This adds an `initialStatus` field to the `Comment`. It also adds `status` to `CommentRevision`.

## Does this PR introduce any new environment variables or feature flags?

no

## If any indexes were added, were they added to `INDEXES.md`?

n/a

## How do I test this PR?

You can set a user to pre-mod. Create comments as that user. See that they have the pre-mod label in moderate cards, in moderation queues and in user history drawer. Approve/reject these comments. See that they still have the pre-mod label based on their initial pre-mod status.

### CODE REVIEW

File: client/src/core/client/admin/components/ModerateCard/MarkersContainer.tsx
Comment (nick-funk) [inline client/src/core/client/admin/components/ModerateCard/MarkersContainer.tsx]: Do we have string type enums that could be imported here instead of using raw string comparisons?

File: server/src/core/server/models/comment/comment.ts
Comment (nick-funk) [inline server/src/core/server/models/comment/comment.ts:116]: Thoughts on storing/exposing this in the comment revisions? We follow a history based structure everywhere else in the server, might be cool to have the status per-revision?

**Review by nick-funk — COMMENTED** [review]
Curious what your thoughts are with respect to storing this in the `CommentRevision` instead of an `initialStatus` field on the comment.

**Review by nick-funk — APPROVED** [review]
I'm approving because technically `status` is also a persisted computed state that is calculated from the mod actions... so this `intialStatus` makes sense in that regard too...

Coral is so weird... I'll leave it up to you which style you prefer cause Coral seems to do both...

File: server/src/core/server/models/comment/comment.ts
Comment (kabeaty) [reply server/src/core/server/models/comment/comment.ts:116]: Yeah, I like this idea so that we have access to it for each revision this way. Updated to then use the first revision status to populate `initialStatus` for comments then.

File: client/src/core/client/admin/components/ModerateCard/MarkersContainer.tsx
Comment (kabeaty) [reply client/src/core/client/admin/components/ModerateCard/MarkersContainer.tsx]: Yep, updated to use that in all places status is referenced in this component...thanks!

Comment (nick-funk) [reply client/src/core/client/admin/components/ModerateCard/MarkersContainer.tsx]: awesome, thank you!

File: server/src/core/server/models/comment/comment.ts
Comment (nick-funk) [reply server/src/core/server/models/comment/comment.ts:116]: amazing, thank you!

**Review by nick-funk — APPROVED** [review]
Updates look great, thank you once again! Approved!

**Pull request merged**
